pages/checkin.tsx

```diff
@@ -49,6 +49,19 @@ export default function CheckinPage() {
     address: prefillAddress || guestInfo.data?.address,
   }
 
+  React.useEffect(() => {
+    if (!areaInfo?.data?.frontendUrl) return
+
+    const url = new URL(areaInfo.data.frontendUrl)
+
+    if (window.location.origin === areaInfo.data.frontendUrl) return
```

Comment:
We need to be super sure that `areaInfo.data.frontendUrl` doesn't end with a slash, otherwise we have an unnecessary redirect. Maybe with a validation in the API, or at least a hint in the admin ui?

Reply:
Dont you think there is a clean way to do this in the frontend?

Reply:
Since you already have the `url` object, you can do

```
if (window.location.origin === url.origin)
```